vowpalwabbit/core/include/vw/core/array_parameters_sparse.h

```diff
@@ -20,7 +20,7 @@ class sparse_parameters;
 namespace details
 {
 
-using weight_map = std::unordered_map<uint64_t, std::shared_ptr<VW::weight>>;
+using weight_map = std::map<uint64_t, std::shared_ptr<VW::weight>>;
```

Comment:
What's the reasoning behind making this a map?

Reply:
its showing better latency for some tests, but its primarily from trying to predict on with features that aren't in the model, so it's unclear if this is an optimization

Reply:
Max Latencies
Old: 17506
New: 8642
New ordered: 4168
 
Average Latencies
Old: 234.6244564186664
New: 279.0359539029642
New ordered: 192.9708312415632

These were the results for one dataset where there were lots of features in the dataset not present in the model.

Reply:
This change has been moved to #4654